### ISSUE TITLE
Smoothie moving slow and multiple decimals in S command

### ISSUE THREAD

**Issue by cojarbi** [openhardwarecoza/LaserWeb#98]
since grbl is not working im back at smoothie.

SVG works without issues ( should add Z movement option )

CALIBRATION: ( 2000 FR)

Movement is very slow no matter whats setting for feedrate
Power is Off all the time
17 decimals in S command

Attached gcode
[smoothie calibration GCODE-LaserWeb (2).gcode.TXT](https://github.com/openhardwarecoza/LaserWeb/files/113367/smoothie.calibration.GCODE-LaserWeb.2.gcode.TXT)
code for calibration

RASTER: seems to be ok

**Comment by funinthefalls** [comment]
Doesn't seem to be a problem with svg2gcode.js. The grbl and marlin code appears to be the same.

From svg2code.js:
  // Firmware Specific Gcode Output
    if (firmware.indexOf('Grbl') == 0) {
        intensity = svglaserpwr.map(0, 100, 0, 255);
        //console.log('Mapping Intensity range for Grbl S0-S255');
        intensity = intensity.toFixed(0);
}
 else {
        intensity = svglaserpwr.map(0, 100, 0, 100);
        //console.log('Mapping Intensity range for S0-S100');
        intensity = intensity.toFixed(0);
    }

Where else do you post-process the Gcode?

**Comment by cojarbi** [comment]
For smoothie no where else. Also i can say pretty surely that the power values are off.
Anything bellow 80% is almost 0%
Attached gcode of test. This was done from an svg file with multiple lines using color assignment. 10 colores 10 power setting, same 500 feed rate.
[drawing2-LaserWeb.gcode.txt](https://github.com/openhardwarecoza/LaserWeb/files/113422/drawing2-LaserWeb.gcode.txt)

[drawing2 copy.svg.txt](https://github.com/openhardwarecoza/LaserWeb/files/113423/drawing2.copy.svg.txt)
![20160202_000059-1](https://cloud.githubusercontent.com/assets/16444438/12740360/72143260-c940-11e5-9d95-ecda3ac83aae.jpg)

Pause button does not respond. Only after pressing clear is that it stops.

**Comment by funinthefalls** [comment]
OOps, missed that you were dealing with smoothie now, not grbl.

else if (firmware.indexOf('Smooth') == 0) {
        intensity = svglaserpwr.map(0, 100, 0, 1);
        //console.log('Mapping Intensity range for Smoothieware S0-S1');
        intensity = intensity.toFixed(2);
    }

setting the intensity seems to be different for smoothie, all the others use "intensity = intensity.**toFixed(0)**;" and smoothie uses "intensity = intensity.**toFixed(2)**

Could this be the problem?

**Comment by funinthefalls** [comment]
Also in the raster section it sets all the other firmware to "**intensity.toFixed(0)**" but sets smoothie to "**intensity.toFixed(1)**"
Was setting "**intensity = intensity.toFixed(2)**" meant to be "**intensity = intensity.toFixed(1)**"

**Comment by ghost** [comment]
Smoothie, being a 32 bit board, isnt limeted to 255 levels of gray (;
S0 - S1 can even do 3 decimals.  We set it to 1 to only have 10 levels - makes the optimisation very fast (albeit at 10 gray levels instead of full grayscale)

**Comment by cojarbi** [comment]
Ok so that's not an issue then.  But why I'm getting no power bellow 80%? There must be something specific to smoothie since it works in grbl

**Comment by ghost** [comment]
According to http://www.shapeoko.com/forum/viewtopic.php?t=7559&p=59718 the Jtech lasers need a specific PWM frequency.  

In your smoothie config, what is your:

`#laser_module_pwm_period                      20              # this sets the pwm frequency as the period in microseconds`

**Comment by ghost** [comment]
Looks like it needs 5khz (http://jtechphotonics.com/?page_id=126)
20 microseconds = 500khz?

**Comment by cojarbi** [comment]
tested now is completely off. no power at all. So this means we need to go bellow 20?

**Comment by ghost** [comment]
Longer microseconds will mean lower frequency.  So try 200?   See https://www.unitjuggler.com/convert-frequency-from-Hz-to-%C2%B5s(p).html?val=5000

**Comment by cojarbi** [comment]
Disregard my last statement. Something else is wrong. Now not even 20 is working. Last night it did. So I must have another issue. Will troubleshoot and restart testing.

**Comment by cojarbi** [comment]
So 200 seems to be it. Should we add something to the wiki

**Comment by ghost** [comment]
Yes please.  Add a page for JTech

**Comment by cojarbi** [comment]
![calibration smoothie](https://cloud.githubusercontent.com/assets/16444438/12815832/e61e3fe8-cb16-11e5-8ef6-2c57f1f76259.jpg)
